src/limits.rs

```diff
@@ -0,0 +1,70 @@
+//! Run-Time Limits
+
+use libc::c_int;
+
+use ffi;
+pub use ffi::Limit;
+
+use Connection;
+
+impl Connection {
+    /// Returns the current value of a limit.
+    pub fn get_limit(&self, limit: Limit) -> i32 {
+        let c = self.db.borrow();
+        unsafe {
+            ffi::sqlite3_limit(c.db(), limit as c_int, -1)
+        }
+    }
+    /// Changes the limit to `new_val`.
+    /// And returns the prior value of the limit.
+    pub fn set_limit(&self, limit: Limit, new_val: i32) -> i32 {
+        let c = self.db.borrow_mut();
+        unsafe {
+            ffi::sqlite3_limit(c.db(), limit as c_int, new_val)
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use ffi::Limit;
+    use Connection;
+
+    #[test]
+    fn test_limit() {
+        let db = Connection::open_in_memory().unwrap();
+        assert_eq!(2147483645, db.set_limit(Limit::SQLITE_LIMIT_LENGTH, 1024));
```

Comment:
I'm not sure these tests are universally applicable; folks might be linking against a system-provided SQLite that set different default limits. Maybe all of these should set first (unchecked) then confirm the get returns the same value? Would be happier to keep these checks but am concerned about failing tests incorrectly; not sure if there's a middle ground.